wire-ios-request-strategy/Sources/Request Strategies/MLS/Actions/ClaimMLSKeyPackageActionHandler.swift

```diff
@@ -68,8 +68,9 @@ class ClaimMLSKeyPackageActionHandler: ActionHandler<ClaimMLSKeyPackageAction> {
             let keyPackagesExcludingSelfClient = payload.keyPackages.filter {
                 $0.client != action.excludedSelfClientId
             }
-
-            guard keyPackagesExcludingSelfClient.isNonEmpty else {
+            let selfUser = ZMUser.selfUser(in: context)
+            let isSelfUserRequesting = selfUser.remoteIdentifier == action.userId && selfUser.domain == action.domain
+            guard isSelfUserRequesting || keyPackagesExcludingSelfClient.isNonEmpty else {
```

Comment:
Why if we're requesting key packages for the self user and none are found, then we consider it a success?

Reply:
@typfel

Reply:
Because the self user not having key packages is not an error.

For other users claiming key packages is successful if there's at least one MLS client with key packages available. This isn't true when claiming key packages for the self user, it would throw `.emptyKeyPackages` unless the self user had at least two MLS clients since we exclude the self client when claiming key packages.

We can deal with it in two ways:
1. Allow returning an empty list of key packages for the self user
2. Catch `.emptyKeyPackages` and check if it was for the self user and then handle the special case

I don't have a strong opinion on how we do it.

Reply:
Right, so it's for the case when the self user with a single mls client creates a group with no one else.